discovery-infra/test_infra/controllers/node_controllers/libvirt_controller.py

```diff
@@ -61,8 +61,14 @@ def start_node(self, node_name):
         node = self.libvirt_connection.lookupByName(node_name)
 
         if not node.isActive():
-            node.create()
-            self._wait_till_domain_has_ips(node)
+            try:
+                node.create()
+                self._wait_till_domain_has_ips(node)
+            except waiting.exceptions.TimeoutExpired:
+                 logging.warning("Node %s failed to recive IP, retrying", node_name)
+                 self.shutdown_node(node_name)
```

Comment:
Assuming node couldn't start at all, would shutdown_node fail?

Reply:
no, shutdown_node checks if the node is active:
`def shutdown_node(self, node_name):
        logging.info("Going to shutdown %s", node_name)
        node = self.libvirt_connection.lookupByName(node_name)

        if node.isActive():
            node.destroy()
`